crates/core/src/config.rs

```diff
@@ -93,8 +122,39 @@ pub struct LogConfig {
     pub directives: Vec<String>,
 }
 
-impl ConfigFile {
-    pub fn read(path: &ConfigToml) -> anyhow::Result<Option<Self>> {
-        parse_config(path.as_ref())
+/// Patch the value of a key in a TOML document,
+/// preserving the formatting and comments of the original value.
+fn patch_value(item: Option<&Item>, value: Option<&str>) -> Option<Item> {
+    match (value, item) {
+        (Some(value), Some(Item::Value(Value::String(v)))) => {
+            let mut new = Value::String(Formatted::new(value.to_string()));
+            let decor = new.decor_mut();
+            *decor = v.decor().clone();
+            Some(new.into())
+        }
+        (Some(val), _) => Some(val.into()),
+        (None, _) => None,
```

Comment:
nit: the caller is just going to match on the returned value anyway. The only time it's `None` is if `value` was `None`, so it seems easier to have the caller match on `value` before calling this, instead of the return value after calling this. Then it would be `fn patch_value(item: Option<&item>, value: &str) -> Item`, which seems a little simpler.